subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/server/sftp/SFTPServer.groovy

```diff
@@ -105,7 +105,7 @@ class SFTPServer extends ServerWithExpectations implements RepositoryServer {
 
     public void restart() {
         stop(true)
-        before()
+        before(port)
```

Comment:
This seems a bit separate to the purpose of the PR. I'm a little worried that there's risk of another process binding to that port while the server is restarting. Do you that concern has any validity?